lib/chef/knife/vsphere_vm_network_add.rb

```diff
@@ -50,19 +60,65 @@ def run
     else fail
     end
 
-    vm.ReconfigVM_Task(
-      spec: {
-        deviceChange: [{
-          operation: :add,
-          fileOperation: nil,
-          device: RbVmomi::VIM::VirtualVmxnet3(
-            key: -1,
-            deviceInfo: { summary: summary, label: '' },
-            backing: backing,
-            addressType: 'generated'
-          )
-        }]
-      }
-    ).wait_for_completion
+    adapter_type = ''
+    if config[:adapter_type] == 'vmxnet3'
```

Comment:
A small thing, but there's a helper method called `get_config` that helps debugging/support and we'd prefer if you used that rather than the config hash, e.g. `get_config(:adapter_type)`.